src/cdn_definitions/schema.yaml

```diff
@@ -11,7 +11,192 @@ $id: http://release-engineering.github.io/cdn-definitions/schema.json
 
 
 definitions:
+  arch:
+    type: string
+
+  arch_list:
+    type: array
+    items:
+      $ref: "#/definitions/arch"
+    uniqueItems: true
+
+  release_version:
+    type: string
+    pattern: "^[0-9]+\\.[0-9]+$"
+
+  major_version:
+    type: integer
+
+  major_version_string:
+    type: string
+    pattern: "^[0-9]+$"
+
+  signing_key:
+    type: string
+    pattern: "^[A-Fa-f0-9]{8}"
+
+  product_id:
+    type: string
+    pattern: "^[0-9]+$"
+
+
+  product_id_list:
+    type: array
+    items:
+      $ref: "#/definitions/product_id"
+    uniqueItems: true
+
+  signing_key_list:
+    type: array
+    items:
+      $ref: "#/definitions/signing_key"
+    uniqueItems: true
+
+  release_version_list:
+    type: array
+    items:
+      $ref: "#/definitions/release_version"
+    uniqueItems: true
+
+  major_or_release_version:
+    type: string
+    pattern: "^[0-9]+(\\.[0-9]+)*$"
+
+  # various product definition mappings
+  version_to_arch_list_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/major_version_string"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/arch_list"
+    additionalProperties: false
+
+  major_to_release_version_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/major_version_string"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/release_version"
+    additionalProperties: false
+
+  release_stream_mapping:
+    type: object
+    properties:
+      aus:
+        $ref: "#/definitions/release_version_list"
+      eus:
+        $ref: "#/definitions/release_version_list"
+      tus:
+        $ref: "#/definitions/release_version_list"
+      e4s:
+        $ref: "#/definitions/release_version_list"
+    requiredProperties:
+    - aus
+    - eus
+    - tus
+    - e4s
+    patternProperties:
+      ".*":
```

Comment:
A question, since this .* will already match all the existing eus, tus etc, is it necessary to list those explicitly?  i.e. if you just delete the "properties" block, does it already work with same semantics.

Reply:
No, it seems that I do not need to list them explicitly.

I have reconsidered explicitly requiring any of the streams. This is another requirement that could instead be enforced via a test in cdn-defintions-private. If we were to add an additional stream to cdn-definitions-private and wanted to avoid disclosing this information to the public, we'd then have to worry about adding that stream to the "required" list of update streams in cdn-definitions after the release was announced publicly. I think a test would serve us just as well, and would be less annoying to update.

I'll leave the four streams in data.yaml as an example, but I will remove the streams from the list of required properties.